packages/x-components/tests/unit/sliding-panel.spec.ts

```diff
@@ -0,0 +1,304 @@
+import { mount } from '@cypress/vue';
+import 'reflect-metadata';
+import Vue from 'vue';
+import { mockedAdapter } from '../../src/adapter/mocked-adapter';
+import SlidingPanel from '../../src/components/sliding-panel.vue';
+import { BaseXBus } from '../../src/plugins/x-bus';
+import { XPlugin } from '../../src/plugins/x-plugin';
+
+/**
+ * Renders an {@link SlidingPanel} component with the provided options.
+ *
+ * @param  options - Options to render {@link SlidingPanel} with.
+ * @returns Helper methods for the rendered {@link SlidingPanel}.
+ */
+function renderSlidingPanel({
+  items = 8,
+  scrollFactor,
+  resetOnContentChange,
+  showButtons,
+  slidingPanelWidthPx = 400,
+  itemWidthPx = 100,
+  template = `
+      <div class="wrapper">
+        <div><input v-model.number="items" data-test="items-count" /></div>
+        <SlidingPanel v-bind="$attrs">
+          <ul class="container">
+            <li class="item" v-for="item in items" :key="item">Item {{ item }}</li>
+          </ul>
+        </SlidingPanel>
+      </div>
+  `
+}: RenderSlidingPanelOptions = {}): RenderSlidingPanelAPI {
+  XPlugin.resetInstance();
+
+  mount(
+    {
+      components: {
+        SlidingPanel
+      },
+      inheritAttrs: false,
+      template,
+      data() {
+        return {
+          items
+        };
+      }
+    },
+    {
+      vue: Vue.extend({}),
+      plugins: [[new XPlugin(new BaseXBus()), { adapter: mockedAdapter }]],
+      propsData: {
+        scrollFactor,
+        resetOnContentChange,
+        showButtons
+      },
+      style: `
+        .wrapper {
+          display: inline-flex;
+          flex-flow: column nowrap;
+        }
+        .x-sliding-panel__scroll {
+          flex: 0 0 auto !important;
+          width: ${slidingPanelWidthPx}px;
+        }
+
+        .container {
+          display: flex;
+          flex-flow: row nowrap;
+          padding: 0;
+          list-style: none;
+        }
+
+        .item {
+          flex: 0 0 auto;
+          width: ${itemWidthPx}px;
+        }`
+    }
+  );
+
+  return {
+    getSlidingPanel() {
+      return cy.getByDataTest('sliding-panel');
+    },
+    getLeftButton() {
+      return cy.getByDataTest('sliding-panel-left-button');
+    },
+    getRightButton() {
+      return cy.getByDataTest('sliding-panel-right-button');
+    },
+    getScrollPosition() {
+      return cy.getByDataTest('sliding-panel-scroll').invoke('scrollLeft');
```

Comment:
Why a función which names implies getting the position of the scroll is a call to a method which mades a scroll and also returns void?
```javascript
protected scrollLeft(): void {
  this.scrollTo(-this.$refs.scrollContainer.clientWidth);
}
```

Where this position comes?